realm/realm-library/build.gradle

```diff
@@ -44,6 +44,8 @@ dependencies {
     androidTestCompile 'com.android.support:support-annotations:23.1.1'
     androidTestCompile 'com.android.support.test:runner:0.4.1'
     androidTestCompile 'com.android.support.test:rules:0.4.1'
+    androidTestCompile 'com.google.dexmaker:dexmaker:1.2'
+    androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'
```

Comment:
What Christian said. Well... not really manipulation, that's not possible to do on Android at runtime, but it can create new classes.